### ISSUE TITLE
One Dark Syntax colors suddenly changed

### ISSUE THREAD

**Issue by jacobdcastro** [atom/atom#18307]
Since updating to Atom 1.32, my favorite syntax theme (One Dark) colors have changed. This is in regards to JavaScript (mainly, React). The actual colors didn't change to different shades (the blue is still the same blue, red is still the same red, etc.). But the syntax highlighting has changed to where what terms used to be a certain color, are a different color now. Here's what I'm seeing on my code:

**Ignore all ESlint highlighting**

![import-section1](https://user-images.githubusercontent.com/15962057/47402248-7e5c9e80-d6f9-11e8-8676-136f2e4d41fa.png)
- Here, all imported components used to be red. 
- The `styled.div` in the styled components used to be red as well
- The CSS string used to replicate normal CSS syntax (property = white, value = orange/white)

![constructor-method1](https://user-images.githubusercontent.com/15962057/47402503-cc25d680-d6fa-11e8-92c3-abb700395369.png)
- `props` and `this` and all other first objects were red

![render-method1](https://user-images.githubusercontent.com/15962057/47402644-7867bd00-d6fb-11e8-9319-85fe04281a47.png)
- Component tags in JSX used to be yellow, all normal HTML-type tags red
- variables used to be red
- In JSX, the `{}` Javascript insert brackets used to be red

![graphql-query](https://user-images.githubusercontent.com/15962057/47402869-2c694800-d6fc-11e8-9ca2-7b5813e3ebce.png)
- GraphQL queries used to replicate normal object colors

These were all that changed on my UI since I updated Atom to 1.32.0. It was normal, then I updated, and came back to this. I liked the syntax colors before the update. Unsure of why they changed. Would love to revert back to old colors!

Thanks!

**Comment by Aerijo** [comment]
It was changed because Tree-sitter parsers have been enabled by default. There are still some differences between the original (TextMate) and the new grammars. It's normally possible to add back the old scopes (or they may be there already under a different name), so you can try raising an issue with what you want added on the corresponding `language-*` repo.

Moving forward, Tree-sitter parsers are expected to provide better information for packages to take advantage of, as well as improved highlighting speed.

To revert, toggle `Settings -> Core -> Use Tree Sitter Parsers`. This should not be a problem now, as they are still quite young and not many packages rely on them.

**Comment by nordboerg** [comment]
this change in the syntax theme is definitely an issue for a lot of developers. this theme is in use for a long long time and this is just too much change at once which is very bad for user experience.

**Comment by Aerijo** [comment]
@nordboerg This isn't a change in the theme, it's a change in the underlying grammars that provide the information for the theme to act on.

As I also said, you can raise issues if some highlighting has been ommitted. Some changes are intentional, and others can be fixed. Is there anything in particular you are referring to?

**Comment by nordboerg** [comment]
@Aerijo thanks for clarifying it.

as @jacobdcastro mentioned it affects almost everything, but in my case the context is typescript; imports, class names, function and variable names, etc.

**Comment by maxbrunsfeld** [comment]
👋 @jacobdcastro some of these changes were intentional, and some weren't.

> Here, all imported components used to be red

This was an intentional change, because the old behavior caused the imported value to appear *first* in red (at the site of the import) and then later, in a completely different color, which is counterproductive with regard to code comprehension. A given 'entity' should ideally appear in a consistent color throughout the file. This isn't possible 100% of the time, but we can do our best.

> `props` and `this` and all other first objects were red

Similar to the import case, I consider this a bug in the old system. Variables like `props` used to appear in different colors in different contexts: red when they were followed by a `.`, and white otherwise.

> variables used to be red

In general, variables weren't red, they were white. Certain variables would *sometimes* appear red, like if they were followed by a `.`.

> The styled.div in the styled components used to be red as well

I also would submit that this is an improvement. The old system was unable to recognize that `div` is a function in that context, so it colored it red, the color of object properties. The new system correctly recognizes that `div` is a function that's being invoked with a template literal, and colors it blue, the color of functions.

> The CSS string used to replicate normal CSS syntax (property = white, value = orange/white)

This was a known issue that we plan to fix. We didn't have time to get CSS parsing into the 1.32 release, but it's in progress here: https://github.com/atom/language-css/pull/156.

> Component tags in JSX used to be yellow, all normal HTML-type tags red
> In JSX, the {} Javascript insert brackets used to be red

These are both mistakes on my part. I will correct them.

> GraphQL queries used to replicate normal object colors

Wow, I didn't realize we did this by default; I thought this would only happen if you had a GraphQL package installed. I will correct this as well.

**Comment by fishcharlie** [comment]
> In general, variables weren't red, they were white. Certain variables would sometimes appear red, like if they were followed by a `.`.

@maxbrunsfeld Is that intentional, or something that will be fixed? Will variables followed by a `.` be reverted back to red or will they stay white?

**Comment by maxbrunsfeld** [comment]
That one was definitely an intentional change.

Coloring variables red when they're followed by a `.` violates what I think is a pretty important property of syntax highlighting - a given entity should appear in a consistent color each time it appears.

Secondly, the classes that *caused* it to be red, `variable.other.object` did not make sense. Being followed by a `.` does not mean that something is an object in JS. Example:

```js
var x = 'not-an-object';
x.length;
```

Finally, in many themes (e.g. `One Light` and `One Dark`), that coloring resulted in the variable being shown in the *same* color as the property name (red), which detracts from the meaning of that color.

**Comment by pbarbiero** [comment]
Not sure if this belongs here, but I am also seeing significant change. I have a lot more white showing up in my chained calls, and its hard to see whats a variable vs a call.

(Syntax is no-caffeine)

Before:
![before](https://user-images.githubusercontent.com/20247733/47457906-f381af80-d79d-11e8-8da8-ee1680fc4ba2.png)

After: 
![after](https://user-images.githubusercontent.com/20247733/47457909-f54b7300-d79d-11e8-8311-9bea630b00e3.png)

**Comment by terribedore** [comment]
I'm seeing my comment tags (TODO, BUG, etc) are no longer highlighted (used to be a magenta/pink just yesterday - I use 'One Dark' ui and syntax). I can bring them back by using @Aerijo's 'hack' of disabling the tree sitter parsers.
![tree sitter parser enabled](https://user-images.githubusercontent.com/33574428/47460965-457a0380-d7a5-11e8-929a-9f4a6eadf893.png)
![tree sitter parser disabled](https://user-images.githubusercontent.com/33574428/47460970-4743c700-d7a5-11e8-90ac-8b47e05444e8.png)

**Comment by Arcanemagus** [comment]
@pbarbiero Much of that has to do with your chosen syntax theme, not the parsing system in use.

Code:
```js
const doStuff = ({ var1, var2, var3 = null }) => {
  const stuffToDo = var1.map(x => {
    const newStuff = {
      a: x.a,
      id: x.id,
      body: JSON.stringify(var2),
    };
    if (var3) {
      newStuff.extraId = var3;
    }
    return newStuff;
  });

  return knex
    .insert(stuffToDo)
    .into('table')
    .then(() => true)
    .catch(error => {
      logger.info(error);
      return false;
    });
};
```

TextMate (old system):
![image](https://user-images.githubusercontent.com/427137/47460963-81589d00-d794-11e8-8ca5-d92345a0d232.png)

Tree-sitter (new system):
![image](https://user-images.githubusercontent.com/427137/47461010-99c8b780-d794-11e8-8900-74cee7b5a14a.png)

Note that there _are_ differences, many of them already explained in https://github.com/atom/atom/issues/18307#issuecomment-432738165.

**Comment by Ben3eeE** [comment]
@terribedore The language todo issue is caputred in https://github.com/atom/language-todo/issues/82 and https://github.com/atom/atom/issues/18196

**Comment by pbarbiero** [comment]
@Arcanemagus 

Thanks, that made me investigate further. Looks like it IS a change, there no longer are span wrappings around the parameters in the function call (for example)

Given this line:
`const doStuff = ({ var1, var2, var3 = null }) => {`

I no longer have this wrapping my variables:
```
<span class="syntax--variable syntax--other syntax--readwrite syntax--shorthandpropertyname syntax--js">var1</span>
```

**Comment by Arcanemagus** [comment]
@pbarbiero From those scopes it looks like you have the third party `language-babel` installed and weren't using `language-javascript` in the first place. If you want to keep using `language-babel` for JavaScript files you can easily [customize how Atom recognizes those files](https://flight-manual.atom.io/using-atom/sections/basic-customization/#customizing-language-recognition) by adding the following to your config as described in that link:
```cson
customFileTypes:
  'source.js.jsx': [
    'js'
  ]
```

**Comment by gavinpatkinson** [comment]
It would be great if this sort of thing did not change on it's own.

**Comment by jacobdcastro** [comment]
Yes, turning 'Use Tree Sitter Parsers' to `false` was exactly the issue. My syntax highlighting is 100% back to normal. Here are screenshots of what it always was for me, for future reference! :)

![import-section](https://user-images.githubusercontent.com/15962057/47463358-d7304380-d79a-11e8-8bc0-5976663053bc.png)
![render-method](https://user-images.githubusercontent.com/15962057/47463365-dac3ca80-d79a-11e8-972c-ce2cb13a5f32.png)
![graphql-query](https://user-images.githubusercontent.com/15962057/47463371-ddbebb00-d79a-11e8-90f5-bf9297b94ac3.png)

Thanks all for sharing our input and similar issues! Great to see that developers, especially newer ones like myself, aren't alone and/or crazy! 

Much Love!
Happy coding.

**Comment by pbarbiero** [comment]
Thanks @Arcanemagus -- that fixed my issue. Just for reference however, yes I am using `language-babel` but I do still have the core atom `language-javascript` enabled so I am still a little confused, but thankful for the workaround none-the-less!

**Comment by maxbrunsfeld** [comment]
The JSX highlighting changes have been fixed and will ship in Atom 1.32.1. CSS string highlighting *may* make it into that release, but I'm not sure.

**Comment by roskelld** [comment]
It's amazing how powerful syntax highlighting is. I noticed this change yesterday, first I thought it was some config messup as I saw no indication anything had changed. I tried to just work with it as I needed to focus on getting a job done, but boy was it hard, I felt like I was seeing bugs everywhere in the code.
